From 051b8db7221f308b242ddb72428d9ca10f14757c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Kacper=20Michaj=C5=82ow?= <kasper93@gmail.com>
Date: Tue, 29 Sep 2026 11:34:17 +0200
Subject: [PATCH] avformat/hls: free the subtitle context if copying the
 whitelists fails
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Free it like the other error paths of init_subtitle_context() do.

Signed-off-by: Kacper Michajłow <kasper93@gmail.com>
---
 libavformat/hls.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/libavformat/hls.c b/libavformat/hls.c
index 97755da3a3ed..5c530215e0d0 100644
--- a/libavformat/hls.c
+++ b/libavformat/hls.c
@@ -1950,8 +1950,11 @@ static int init_subtitle_context(struct playlist *pls)
     pls->ctx->io_open  = nested_io_open;
 
     ret = ff_copy_whiteblacklists(pls->ctx, pls->parent);
-    if (ret < 0)
+    if (ret < 0) {
+        avformat_free_context(pls->ctx);
+        pls->ctx = NULL;
         return ret;
+    }
 
     in_fmt = av_find_input_format("webvtt");
     av_dict_copy(&opts, c->seg_format_opts, 0);
